src/java.base/share/classes/java/nio/channels/FileLock.java

```diff
@@ -269,14 +269,34 @@ public final boolean isShared() {
      * @param   size
      *          The size of the lock range
      *
-     * @return  {@code true} if, and only if, this lock and the given lock
-     *          range overlap by at least one byte
+     * @return  {@code true} if this lock and the given lock range overlap
+     *          by at least one byte; {@code false} if {@code size} is
+     *          negative or the lock range does not overlap this lock
      */
     public final boolean overlaps(long position, long size) {
-        if (position + size <= this.position)
-            return false;               // That is below this
-        if (this.position + this.size <= position)
-            return false;               // This is below that
+        if (size < 0)
+            return false;
+
+        try {
+            if (Math.addExact(this.position, this.size) <= position)
+                return false;               // This is below that
+        } catch (ArithmeticException ignored) {
+            // the sum of this.position and this.size overflows the range of
+            // long hence their mathematical sum is greater than position
+        }
```

Comment:
I think it would be a bit clearer if you move the comment "This is below that" to above the try as something "Test if this is below that". Same for the test to see if that is below this.